cfme/infrastructure/datastore.py

```diff
@@ -105,16 +116,16 @@ def get_detail(self, *ident):
             *ident: An InfoBlock title, followed by the Key name, e.g. "Relationships", "Images"
         Returns: A string representing the contents of the InfoBlock's value.
         """
-        if not self._on_detail_page():
-            sel.force_navigate('infrastructure_datastore', context=self._get_context())
+        self.load_details()
         return details_page.infoblock.text(*ident)
 
     def _on_detail_page(self):
         """ Returns ``True`` if on the datastore detail page, ``False`` if not."""
-        return sel.is_displayed(
-            '//div[@class="dhtmlxInfoBarLabel-2"][contains(., "{}") and contains(., "{}")]'.format(
-                self.name, "Summary")
-        )
+        ensure_browser_open()
```

Comment:
We shouldn't ever put this inside an object file, why is this getting raised? A force navigate should always happen somewhere first. We can maybe wrap it in a try except, but putting the ```ensure_browser_open()``` shouldn't be the solution.